pkg/actors/targets/workflow/orchestrator/run.go

```diff
@@ -140,7 +141,47 @@ func (o *orchestrator) runWorkflow(ctx context.Context, reminder *actorapi.Remin
 			return todo.RunCompletedFalse, wferrors.NewRecoverable(todo.ErrExecutionAborted)
 		}
 	}
-	log.Debugf("Workflow actor '%s': workflow execution returned with status '%s' instanceId '%s'", o.actorID, runtimestate.RuntimeStatus(rs).String(), wi.InstanceID)
+	rs = wi.State
+
+	hasPatchMismatch := o.checkPatchMismatch(rs)
+	if hasPatchMismatch {
+		rs.CompletedEvent = nil
+		rs.CompletedTime = nil
```

Comment:
Why is this necessary?

Reply:
By the time we run this, the executor might have set the `CompletedEvent`, but if there has been a mismatch we can't let it be set because otherwise the workflow will be considered `completed` instead of stalled, as the `IsCompleted` uses the presence of this field.